Add @next/vendored

lfades opened the pull request:
This PR aims to replace the use of `next/dist/compiled/*` with `@next/vendored/*`.

## Motivation

- Remove `next/src/compiled` from source. Currently Next.js clones its contents into `next/dist/compiled` but we can run the tasks that generate the modules fast enough with caching.
- The imports to `next/dist/compiled/*` will now to go `@next/vendored/*`, a separate package that has all the third party modules that were compiled by ncc. This allows Next.js to have `exports` inside `package.json` in a future update, which is currently highly difficult due to it having inner node packages (i.e nested modules that are resolved after reading a `package.json`).
- Move all of the related ncc tasks and compiled code out of the `next` repo.

## Status

Some things that have been done so far:

-  A custom task runner is being used instead of `taskr`. The reason is `taskr` doesn't work very well with ESM (The `@taskr/esnext` package is a hack and it's still commonjs), because of this resolving a package always sends its commonjs distribution to ncc. There are also some differences in how the tasks are declared, particularly I wanted to have a way to generate multiple tasks with a loop for the most common cases.
- The custom task runner is quite simple and it can also be made into its own package.

### Todo

- Move all the related ncc tasks from `next` to `@next/vendored`
- Use Turborepo caching like in https://github.com/vercel/next.js/pull/64825

### Testing locally

After running `pnpm install`, build `@next/vendored` with:

```bash
cd packages/next-vendored && pnpm dev
```

and then run the ncc tasks in `@next/vendored`:

```bash
pnpm task-runner ncc
```

Run `pnpm dev` at the root and the Next.js build should work. I'm testing it locally in an app from  create-next-app and with `pnpm next-with-deps`

<!-- Thanks for opening a PR! Your contribution is much appreciated.
To make sure your PR is handled as smoothly as possible we request that you follow the checklist sections below.
Choose the right checklist for the change(s) that you're making:

## For Contributors

### Improving Documentation

- Run `pnpm prettier-fix` to fix formatting issues before opening the PR.
- Read the Docs Contribution Guide to ensure your contribution follows the docs guidelines: https://nextjs.org/docs/community/contribution-guide

### Adding or Updating Examples

- The "examples guidelines" are followed from our contributing doc https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md
- Make sure the linting passes by running `pnpm build && pnpm lint`. See https://github.com/vercel/next.js/blob/canary/contributing/repository/linting.md

### Fixing a bug

- Related issues linked using `fixes #number`
- Tests added. See: https://github.com/vercel/next.js/blob/canary/contributing/core/testing.md#writing-tests-for-nextjs
- Errors have a helpful link attached, see https://github.com/vercel/next.js/blob/canary/contributing.md

### Adding a feature

- Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR. (A discussion must be opened, see https://github.com/vercel/next.js/discussions/new?category=ideas)
- Related issues/discussions are linked using `fixes #number`
- e2e tests added (https://github.com/vercel/next.js/blob/canary/contributing/core/testing.md#writing-tests-for-nextjs)
- Documentation added
- Telemetry added. In case of a feature if it's used or not.
- Errors have a helpful link attached, see https://github.com/vercel/next.js/blob/canary/contributing.md


## For Maintainers

- Minimal description (aim for explaining to someone not on the team to understand the PR)
- When linking to a Slack thread, you might want to share details of the conclusion
- Link both the Linear (Fixes NEXT-xxx) and the GitHub issues
- Add review comments if necessary to explain to the reviewer the logic behind a change

### What?

### Why?

### How?

Closes NEXT-
Fixes #

-->

eps1lon commented:
FYI we've replaced taskr is being replaced with using Turborepo: https://github.com/vercel/next.js/pull/64825

lfades commented:
> FYI we've replaced taskr is being replaced with using Turborepo: #64825

Thanks for sharing it. `taskr` is still being used there but it's being cached by Turborepo. I'll add that to this PR.

Note that I replaced `taskr` to resolve modules for ncc in a ESM environment.

ijjk commented:
## Failing test suites <!--24278951767-->
Commit: 66360784626d0f66caa208a5048b9e8583450a4f

`TURBOPACK=1 pnpm test-start test/e2e/app-dir/app-simple-routes/app-simple-routes.test.ts` (turbopack)
- app-simple-routes > works with simple routes > renders a node route
- app-simple-routes > works with simple routes > renders a edge route

<details>
<summary>Expand output</summary>

  ● app-simple-routes › works with simple routes › renders a node route

    next build failed with code/signal 1

       97 |           if (code || signal)
       98 |             reject(
    >  99 |               new Error(`next build failed with code/signal ${code || signal}`)
          |               ^
      100 |             )
      101 |           else resolve()
      102 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:99:15)

  ● app-simple-routes › works with simple routes › renders a edge route

    next build failed with code/signal 1

       97 |           if (code || signal)
       98 |             reject(
    >  99 |               new Error(`next build failed with code/signal ${code || signal}`)
          |               ^
      100 |             )
      101 |           else resolve()
      102 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:99:15)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

`TURBOPACK=1 pnpm test-start test/e2e/app-dir/app-edge-root-layout/index.test.ts` (turbopack)
- app-dir edge runtime root layout > should not emit metadata files into bad paths
- app-dir edge runtime root layout > should mark static contain metadata routes as edge functions

<details>
<summary>Expand output</summary>

  ● app-dir edge runtime root layout › should not emit metadata files into bad paths

    next build failed with code/signal 1

       97 |           if (code || signal)
       98 |             reject(
    >  99 |               new Error(`next build failed with code/signal ${code || signal}`)
          |               ^
      100 |             )
      101 |           else resolve()
      102 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:99:15)

  ● app-dir edge runtime root layout › should mark static contain metadata routes as edge functions

    next build failed with code/signal 1

       97 |           if (code || signal)
       98 |             reject(
    >  99 |               new Error(`next build failed with code/signal ${code || signal}`)
          |               ^
      100 |             )
      101 |           else resolve()
      102 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:99:15)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

ijjk commented:
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js next-vendored | Change |
| - | - | - | - |
| buildDuration | 17.1s | 15.7s | N/A |
| buildDurationCached | 8.3s | 6.9s | N/A |
| nodeModulesSize | 351 MB | 501 MB | ⚠️ +150 MB |
| nextStartRea..uration (ms) | 407ms | 409ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js next-vendored | Change |
| - | - | - | - |
| 2358-HASH.js gzip | 35.6 kB | 35.5 kB | N/A |
| 2741.HASH.js gzip | 171 B | 170 B | N/A |
| 56e3f737-HASH.js gzip | 51.4 kB | 51.5 kB | N/A |
| 5751-HASH.js gzip | 5.05 kB | 5.04 kB | N/A |
| framework-HASH.js gzip | 56.5 kB | 56.5 kB | N/A |
| main-app-HASH.js gzip | 222 B | 222 B | ✓ |
| main-HASH.js gzip | 32.3 kB | 32.1 kB | N/A |
| webpack-HASH.js gzip | 1.71 kB | 1.65 kB | N/A |
| Overall change | 222 B | 222 B | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js next-vendored | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js next-vendored | Change |
| - | - | - | - |
| _app-HASH.js gzip | 193 B | 193 B | ✓ |
| _error-HASH.js gzip | 192 B | 191 B | N/A |
| amp-HASH.js gzip | 511 B | 506 B | N/A |
| css-HASH.js gzip | 342 B | 343 B | N/A |
| dynamic-HASH.js gzip | 2.53 kB | 2.52 kB | N/A |
| edge-ssr-HASH.js gzip | 266 B | 264 B | N/A |
| head-HASH.js gzip | 364 B | 365 B | N/A |
| hooks-HASH.js gzip | 392 B | 390 B | N/A |
| image-HASH.js gzip | 4.27 kB | 4.27 kB | N/A |
| index-HASH.js gzip | 267 B | 268 B | N/A |
| link-HASH.js gzip | 2.69 kB | 2.69 kB | N/A |
| routerDirect..HASH.js gzip | 327 B | 328 B | N/A |
| script-HASH.js gzip | 396 B | 397 B | N/A |
| withRouter-HASH.js gzip | 323 B | 325 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 299 B | 299 B | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js next-vendored | Change |
| - | - | - | - |
| _buildManifest.js gzip | 486 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js next-vendored | Change |
| - | - | - | - |
| index.html gzip | 523 B | 521 B | N/A |
| link.html gzip | 535 B | 536 B | N/A |
| withRouter.html gzip | 519 B | 517 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js next-vendored | Change |
| - | - | - | - |
| edge-ssr.js gzip | 124 kB | 125 kB | N/A |
| page.js gzip | 188 kB | 188 kB | ⚠️ +195 B |
| Overall change | 188 kB | 188 kB | ⚠️ +195 B |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js next-vendored | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 663 B | 659 B | N/A |
| middleware-r..fest.js gzip | 158 B | 156 B | N/A |
| middleware.js gzip | 29.5 kB | 29.4 kB | N/A |
| edge-runtime..pack.js gzip | 1.02 kB | 1.02 kB | ✓ |
| Overall change | 1.02 kB | 1.02 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js next-vendored | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 185 kB | 185 kB | N/A |
| app-page-exp..prod.js gzip | 112 kB | 112 kB | N/A |
| app-page-tur..prod.js gzip | 123 kB | 123 kB | N/A |
| app-page-tur..prod.js gzip | 119 kB | 119 kB | N/A |
| app-page.run...dev.js gzip | 177 kB | 177 kB | N/A |
| app-page.run..prod.js gzip | 109 kB | 108 kB | N/A |
| app-route-ex...dev.js gzip | 24.5 kB | 24.5 kB | N/A |
| app-route-ex..prod.js gzip | 18.3 kB | 18.3 kB | N/A |
| app-route-tu..prod.js gzip | 18.3 kB | 18.3 kB | N/A |
| app-route-tu..prod.js gzip | 18.1 kB | 18.1 kB | N/A |
| app-route.ru...dev.js gzip | 24.2 kB | 24.2 kB | N/A |
| app-route.ru..prod.js gzip | 18.1 kB | 18.1 kB | N/A |
| pages-api-tu..prod.js gzip | 9.55 kB | 9.55 kB | N/A |
| pages-api.ru...dev.js gzip | 9.82 kB | 9.82 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.55 kB | 9.54 kB | N/A |
| pages-turbo...prod.js gzip | 21.5 kB | 21.5 kB | N/A |
| pages.runtim...dev.js gzip | 22 kB | 22 kB | N/A |
| pages.runtim..prod.js gzip | 21.5 kB | 21.5 kB | N/A |
| server.runti..prod.js gzip | 52.1 kB | 52 kB | N/A |
| Overall change | 9.82 kB | 9.82 kB | ✓ |

</details>

<details>
<summary><strong>build cache</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js next-vendored | Change |
| - | - | - | - |
| 0.pack gzip | 1.67 MB | 1.67 MB | N/A |
| index.pack gzip | 129 kB | 225 kB | ⚠️ +96.2 kB |
| Overall change | 129 kB | 225 kB | ⚠️ +96.2 kB |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>page.js</strong></summary>

```diff
@@ -15,7 +15,7 @@
       /***/
     },
 
-    /***/ 3080: /***/ (
+    /***/ 2444: /***/ (
       __unused_webpack_module,
       __webpack_exports__,
       __webpack_require__
@@ -30,7 +30,7 @@
         default: () => /* binding */ nHandler,
       });
 
-      // NAMESPACE OBJECT: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/build/webpack/loaders/next-app-loader.js?name=app%2Fapp-edge-ssr%2Fpage&page=%2Fapp-edge-ssr%2Fpage&pagePath=private-next-app-dir%2Fapp-edge-ssr%2Fpage.js&appDir=%2Ftmp%2Fnext-stats9OIWLP%2Fstats-app%2Fapp&appPaths=%2Fapp-edge-ssr%2Fpage&pageExtensions=tsx&pageExtensions=ts&pageExtensions=jsx&pageExtensions=js&basePath=&assetPrefix=&nextConfigOutput=&preferredRegion=&middlewareConfig=e30%3D!./app/app-edge-ssr/page.js?__next_edge_ssr_entry__
+      // NAMESPACE OBJECT: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/build/webpack/loaders/next-app-loader.js?name=app%2Fapp-edge-ssr%2Fpage&page=%2Fapp-edge-ssr%2Fpage&pagePath=private-next-app-dir%2Fapp-edge-ssr%2Fpage.js&appDir=%2Ftmp%2Fnext-stats9OIWLP%2Fstats-app%2Fapp&appPaths=%2Fapp-edge-ssr%2Fpage&pageExtensions=tsx&pageExtensions=ts&pageExtensions=jsx&pageExtensions=js&basePath=&assetPrefix=&nextConfigOutput=&preferredRegion=&middlewareConfig=e30%3D!./app/app-edge-ssr/page.js?__next_edge_ssr_entry__
       var page_next_edge_ssr_entry_namespaceObject = {};
       __webpack_require__.r(page_next_edge_ssr_entry_namespaceObject);
       __webpack_require__.d(page_next_edge_ssr_entry_namespaceObject, {
@@ -71,24 +71,24 @@
         tree: () => tree,
       });
 
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/esm/server/web/globals.js
-      var globals = __webpack_require__(6034);
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/esm/server/web/adapter.js + 3 modules
-      var adapter = __webpack_require__(7565);
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/esm/build/webpack/loaders/next-edge-ssr-loader/render.js + 88 modules
-      var render = __webpack_require__(6058);
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/esm/server/lib/incremental-cache/index.js + 3 modules
-      var incremental_cache = __webpack_require__(3656);
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/esm/server/app-render/app-render.js + 50 modules
-      var app_render = __webpack_require__(9984);
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/esm/server/future/route-modules/app-page/module.compiled.js
-      var module_compiled = __webpack_require__(1506);
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/esm/server/future/route-kind.js
-      var route_kind = __webpack_require__(7216);
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/esm/client/components/error-boundary.js
-      var error_boundary = __webpack_require__(2561);
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/esm/server/app-render/entry-base.js + 10 modules
-      var entry_base = __webpack_require__(4134); // CONCATENATED MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/build/webpack/loaders/next-app-loader.js?name=app%2Fapp-edge-ssr%2Fpage&page=%2Fapp-edge-ssr%2Fpage&pagePath=private-next-app-dir%2Fapp-edge-ssr%2Fpage.js&appDir=%2Ftmp%2Fnext-stats9OIWLP%2Fstats-app%2Fapp&appPaths=%2Fapp-edge-ssr%2Fpage&pageExtensions=tsx&pageExtensions=ts&pageExtensions=jsx&pageExtensions=js&basePath=&assetPrefix=&nextConfigOutput=&preferredRegion=&middlewareConfig=e30%3D!./app/app-edge-ssr/page.js?__next_edge_ssr_entry__
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/esm/server/web/globals.js
+      var globals = __webpack_require__(210);
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/esm/server/web/adapter.js + 3 modules
+      var adapter = __webpack_require__(9203);
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/esm/build/webpack/loaders/next-edge-ssr-loader/render.js + 88 modules
+      var render = __webpack_require__(703);
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/esm/server/lib/incremental-cache/index.js + 3 modules
+      var incremental_cache = __webpack_require__(3392);
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/esm/server/app-render/app-render.js + 50 modules
+      var app_render = __webpack_require__(7982);
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/esm/server/future/route-modules/app-page/module.compiled.js
+      var module_compiled = __webpack_require__(3572);
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/esm/server/future/route-kind.js
+      var route_kind = __webpack_require__(3923);
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/esm/client/components/error-boundary.js
+      var error_boundary = __webpack_require__(7434);
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/esm/server/app-render/entry-base.js + 10 modules
+      var entry_base = __webpack_require__(7276); // CONCATENATED MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/build/webpack/loaders/next-app-loader.js?name=app%2Fapp-edge-ssr%2Fpage&page=%2Fapp-edge-ssr%2Fpage&pagePath=private-next-app-dir%2Fapp-edge-ssr%2Fpage.js&appDir=%2Ftmp%2Fnext-stats9OIWLP%2Fstats-app%2Fapp&appPaths=%2Fapp-edge-ssr%2Fpage&pageExtensions=tsx&pageExtensions=ts&pageExtensions=jsx&pageExtensions=js&basePath=&assetPrefix=&nextConfigOutput=&preferredRegion=&middlewareConfig=e30%3D!./app/app-edge-ssr/page.js?__next_edge_ssr_entry__
       // We inject the tree and pages here so that we can use them in the route
       // module.
       const tree = {
@@ -105,7 +105,7 @@
                     page: [
                       () =>
                         Promise.resolve(/* import() eager */).then(
-                          __webpack_require__.bind(__webpack_require__, 2287)
+                          __webpack_require__.bind(__webpack_require__, 8236)
                         ),
                       "/tmp/next-stats9OIWLP/stats-app/app/app-edge-ssr/page.js",
                     ],
@@ -119,14 +119,14 @@
             layout: [
               () =>
                 Promise.resolve(/* import() eager */).then(
-                  __webpack_require__.bind(__webpack_require__, 8369)
+                  __webpack_require__.bind(__webpack_require__, 2716)
                 ),
               "/tmp/next-stats9OIWLP/stats-app/app/layout.js",
             ],
             "not-found": [
               () =>
                 Promise.resolve(/* import() eager */).then(
-                  __webpack_require__.bind(__webpack_require__, 6309)
+                  __webpack_require__.bind(__webpack_require__, 4186)
                 ),
               "next/dist/client/components/not-found-error",
             ],
@@ -162,12 +162,12 @@
       });
 
       //# sourceMappingURL=app-page.js.map
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/esm/lib/page-types.js
-      var page_types = __webpack_require__(8119);
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/esm/server/app-render/encryption-utils.js
-      var encryption_utils = __webpack_require__(3866);
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/esm/server/app-render/action-utils.js
-      var action_utils = __webpack_require__(1679); // CONCATENATED MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/dist/build/webpack/loaders/next-edge-ssr-loader/index.js?{"absolute500Path":"","absoluteAppPath":"next/dist/pages/_app","absoluteDocumentPath":"next/dist/pages/_document","absoluteErrorPath":"next/dist/pages/_error","absolutePagePath":"private-next-app-dir/app-edge-ssr/page.js","dev":false,"isServerComponent":true,"page":"/app-edge-ssr/page","stringifiedConfig":"eyJlbnYiOnt9LCJlc2xpbnQiOnsiaWdub3JlRHVyaW5nQnVpbGRzIjpmYWxzZX0sInR5cGVzY3JpcHQiOnsiaWdub3JlQnVpbGRFcnJvcnMiOmZhbHNlLCJ0c2NvbmZpZ1BhdGgiOiJ0c2NvbmZpZy5qc29uIn0sImRpc3REaXIiOiIubmV4dCIsImNsZWFuRGlzdERpciI6dHJ1ZSwiYXNzZXRQcmVmaXgiOiIiLCJjYWNoZU1heE1lbW9yeVNpemUiOjUyNDI4ODAwLCJjb25maWdPcmlnaW4iOiJuZXh0LmNvbmZpZy5qcyIsInVzZUZpbGVTeXN0ZW1QdWJsaWNSb3V0ZXMiOnRydWUsImdlbmVyYXRlRXRhZ3MiOnRydWUsInBhZ2VFeHRlbnNpb25zIjpbInRzeCIsInRzIiwianN4IiwianMiXSwicG93ZXJlZEJ5SGVhZGVyIjp0cnVlLCJjb21wcmVzcyI6dHJ1ZSwiaW1hZ2VzIjp7ImRldmljZVNpemVzIjpbNjQwLDc1MCw4MjgsMTA4MCwxMjAwLDE5MjAsMjA0OCwzODQwXSwiaW1hZ2VTaXplcyI6WzE2LDMyLDQ4LDY0LDk2LDEyOCwyNTYsMzg0XSwicGF0aCI6Ii9fbmV4dC9pbWFnZSIsImxvYWRlciI6ImRlZmF1bHQiLCJsb2FkZXJGaWxlIjoiIiwiZG9tYWlucyI6W10sImRpc2FibGVTdGF0aWNJbWFnZXMiOmZhbHNlLCJtaW5pbXVtQ2FjaGVUVEwiOjYwLCJmb3JtYXRzIjpbImltYWdlL3dlYnAiXSwiZGFuZ2Vyb3VzbHlBbGxvd1NWRyI6ZmFsc2UsImNvbnRlbnRTZWN1cml0eVBvbGljeSI6InNjcmlwdC1zcmMgJ25vbmUnOyBmcmFtZS1zcmMgJ25vbmUnOyBzYW5kYm94OyIsImNvbnRlbnREaXNwb3NpdGlvblR5cGUiOiJhdHRhY2htZW50IiwicmVtb3RlUGF0dGVybnMiOltdLCJ1bm9wdGltaXplZCI6ZmFsc2V9LCJkZXZJbmRpY2F0b3JzIjp7ImJ1aWxkQWN0aXZpdHkiOnRydWUsImJ1aWxkQWN0aXZpdHlQb3NpdGlvbiI6ImJvdHRvbS1yaWdodCJ9LCJvbkRlbWFuZEVudHJpZXMiOnsibWF4SW5hY3RpdmVBZ2UiOjYwMDAwLCJwYWdlc0J1ZmZlckxlbmd0aCI6NX0sImFtcCI6eyJjYW5vbmljYWxCYXNlIjoiIn0sImJhc2VQYXRoIjoiIiwic2Fzc09wdGlvbnMiOnt9LCJ0cmFpbGluZ1NsYXNoIjpmYWxzZSwiaTE4biI6bnVsbCwicHJvZHVjdGlvbkJyb3dzZXJTb3VyY2VNYXBzIjpmYWxzZSwib3B0aW1pemVGb250cyI6dHJ1ZSwiZXhjbHVkZURlZmF1bHRNb21lbnRMb2NhbGVzIjp0cnVlLCJzZXJ2ZXJSdW50aW1lQ29uZmlnIjp7fSwicHVibGljUnVudGltZUNvbmZpZyI6e30sInJlYWN0UHJvZHVjdGlvblByb2ZpbGluZyI6ZmFsc2UsInJlYWN0U3RyaWN0TW9kZSI6bnVsbCwiaHR0cEFnZW50T3B0aW9ucyI6eyJrZWVwQWxpdmUiOnRydWV9LCJzdGF0aWNQYWdlR2VuZXJhdGlvblRpbWVvdXQiOjYwLCJtb2R1bGFyaXplSW1wb3J0cyI6eyJAbXVpL2ljb25zLW1hdGVyaWFsIjp7InRyYW5zZm9ybSI6IkBtdWkvaWNvbnMtbWF0ZXJpYWwve3ttZW1iZXJ9fSJ9LCJsb2Rhc2giOnsidHJhbnNmb3JtIjoibG9kYXNoL3t7bWVtYmVyfX0ifX0sImV4cGVyaW1lbnRhbCI6eyJmbHlpbmdTaHV0dGxlIjpmYWxzZSwicHJlcmVuZGVyRWFybHlFeGl0Ijp0cnVlLCJzZXJ2ZXJNaW5pZmljYXRpb24iOnRydWUsInNlcnZlclNvdXJjZU1hcHMiOmZhbHNlLCJsaW5rTm9Ub3VjaFN0YXJ0IjpmYWxzZSwiY2FzZVNlbnNpdGl2ZVJvdXRlcyI6ZmFsc2UsInByZWxvYWRFbnRyaWVzT25TdGFydCI6dHJ1ZSwiY2xpZW50Um91dGVyRmlsdGVyIjp0cnVlLCJjbGllbnRSb3V0ZXJGaWx0ZXJSZWRpcmVjdHMiOmZhbHNlLCJmZXRjaENhY2hlS2V5UHJlZml4IjoiIiwibWlkZGxld2FyZVByZWZldGNoIjoiZmxleGlibGUiLCJvcHRpbWlzdGljQ2xpZW50Q2FjaGUiOnRydWUsIm1hbnVhbENsaWVudEJhc2VQYXRoIjpmYWxzZSwiY3B1cyI6MTksIm1lbW9yeUJhc2VkV29ya2Vyc0NvdW50IjpmYWxzZSwiaXNyRmx1c2hUb0Rpc2siOnRydWUsIndvcmtlclRocmVhZHMiOmZhbHNlLCJvcHRpbWl6ZUNzcyI6ZmFsc2UsIm5leHRTY3JpcHRXb3JrZXJzIjpmYWxzZSwic2Nyb2xsUmVzdG9yYXRpb24iOmZhbHNlLCJleHRlcm5hbERpciI6ZmFsc2UsImRpc2FibGVPcHRpbWl6ZWRMb2FkaW5nIjpmYWxzZSwiZ3ppcFNpemUiOnRydWUsImNyYUNvbXBhdCI6ZmFsc2UsImVzbUV4dGVybmFscyI6dHJ1ZSwiZnVsbHlTcGVjaWZpZWQiOmZhbHNlLCJvdXRwdXRGaWxlVHJhY2luZ1Jvb3QiOiIvdG1wL25leHQtc3RhdHM5T0lXTFAvc3RhdHMtYXBwIiwic3djVHJhY2VQcm9maWxpbmciOmZhbHNlLCJmb3JjZVN3Y1RyYW5zZm9ybXMiOmZhbHNlLCJsYXJnZVBhZ2VEYXRhQnl0ZXMiOjEyODAwMCwiYWRqdXN0Rm9udEZhbGxiYWNrcyI6ZmFsc2UsImFkanVzdEZvbnRGYWxsYmFja3NXaXRoU2l6ZUFkanVzdCI6ZmFsc2UsInR5cGVkUm91dGVzIjpmYWxzZSwiaW5zdHJ1bWVudGF0aW9uSG9vayI6ZmFsc2UsInBhcmFsbGVsU2VydmVyQ29tcGlsZXMiOmZhbHNlLCJwYXJhbGxlbFNlcnZlckJ1aWxkVHJhY2VzIjpmYWxzZSwicHByIjpmYWxzZSwid2VicGFja01lbW9yeU9wdGltaXphdGlvbnMiOmZhbHNlLCJvcHRpbWl6ZVNlcnZlclJlYWN0Ijp0cnVlLCJ1c2VFYXJseUltcG9ydCI6ZmFsc2UsInN0YWxlVGltZXMiOnsiZHluYW1pYyI6MCwic3RhdGljIjozMDB9LCJhZnRlciI6ZmFsc2UsIm9wdGltaXplUGFja2FnZUltcG9ydHMiOlsibHVjaWRlLXJlYWN0IiwiZGF0ZS1mbnMiLCJsb2Rhc2gtZXMiLCJyYW1kYSIsImFudGQiLCJyZWFjdC1ib290c3RyYXAiLCJhaG9va3MiLCJAYW50LWRlc2lnbi9pY29ucyIsIkBoZWFkbGVzc3VpL3JlYWN0IiwiQGhlYWRsZXNzdWktZmxvYXQvcmVhY3QiLCJAaGVyb2ljb25zL3JlYWN0LzIwL3NvbGlkIiwiQGhlcm9pY29ucy9yZWFjdC8yNC9zb2xpZCIsIkBoZXJvaWNvbnMvcmVhY3QvMjQvb3V0bGluZSIsIkB2aXN4L3Zpc3giLCJAdHJlbW9yL3JlYWN0IiwicnhqcyIsIkBtdWkvbWF0ZXJpYWwiLCJAbXVpL2ljb25zLW1hdGVyaWFsIiwicmVjaGFydHMiLCJyZWFjdC11c2UiLCJlZmZlY3QiLCJAZWZmZWN0L3NjaGVtYSIsIkBlZmZlY3QvcGxhdGZvcm0iLCJAZWZmZWN0L3BsYXRmb3JtLW5vZGUiLCJAZWZmZWN0L3BsYXRmb3JtLWJyb3dzZXIiLCJAZWZmZWN0L3BsYXRmb3JtLWJ1biIsIkBlZmZlY3Qvc3FsIiwiQGVmZmVjdC9zcWwtbXNzcWwiLCJAZWZmZWN0L3NxbC1teXNxbDIiLCJAZWZmZWN0L3NxbC1wZyIsIkBlZmZlY3Qvc3FsLXNxdWxpdGUtbm9kZSIsIkBlZmZlY3Qvc3FsLXNxdWxpdGUtYnVuIiwiQGVmZmVjdC9zcWwtc3F1bGl0ZS13YXNtIiwiQGVmZmVjdC9zcWwtc3F1bGl0ZS1yZWFjdC1uYXRpdmUiLCJAZWZmZWN0L3JwYyIsIkBlZmZlY3QvcnBjLWh0dHAiLCJAZWZmZWN0L3R5cGVjbGFzcyIsIkBlZmZlY3QvZXhwZXJpbWVudGFsIiwiQGVmZmVjdC9vcGVudGVsZW1ldHJ5IiwiQG1hdGVyaWFsLXVpL2NvcmUiLCJAbWF0ZXJpYWwtdWkvaWNvbnMiLCJAdGFibGVyL2ljb25zLXJlYWN0IiwibXVpLWNvcmUiLCJyZWFjdC1pY29ucy9haSIsInJlYWN0LWljb25zL2JpIiwicmVhY3QtaWNvbnMvYnMiLCJyZWFjdC1pY29ucy9jZyIsInJlYWN0LWljb25zL2NpIiwicmVhY3QtaWNvbnMvZGkiLCJyZWFjdC1pY29ucy9mYSIsInJlYWN0LWljb25zL2ZhNiIsInJlYWN0LWljb25zL2ZjIiwicmVhY3QtaWNvbnMvZmkiLCJyZWFjdC1pY29ucy9naSIsInJlYWN0LWljb25zL2dvIiwicmVhY3QtaWNvbnMvZ3IiLCJyZWFjdC1pY29ucy9oaSIsInJlYWN0LWljb25zL2hpMiIsInJlYWN0LWljb25zL2ltIiwicmVhY3QtaWNvbnMvaW8iLCJyZWFjdC1pY29ucy9pbzUiLCJyZWFjdC1pY29ucy9saWEiLCJyZWFjdC1pY29ucy9saWIiLCJyZWFjdC1pY29ucy9sdSIsInJlYWN0LWljb25zL21kIiwicmVhY3QtaWNvbnMvcGkiLCJyZWFjdC1pY29ucy9yaSIsInJlYWN0LWljb25zL3J4IiwicmVhY3QtaWNvbnMvc2kiLCJyZWFjdC1pY29ucy9zbCIsInJlYWN0LWljb25zL3RiIiwicmVhY3QtaWNvbnMvdGZpIiwicmVhY3QtaWNvbnMvdGkiLCJyZWFjdC1pY29ucy92c2MiLCJyZWFjdC1pY29ucy93aSJdfSwiYnVuZGxlUGFnZXNSb3V0ZXJEZXBlbmRlbmNpZXMiOmZhbHNlLCJjb25maWdGaWxlIjoiL3RtcC9uZXh0LXN0YXRzOU9JV0xQL3N0YXRzLWFwcC9uZXh0LmNvbmZpZy5qcyIsImNvbmZpZ0ZpbGVOYW1lIjoibmV4dC5jb25maWcuanMifQ==","pagesType":"app","appDirLoader":"bmV4dC1hcHAtbG9hZGVyP25hbWU9YXBwJTJGYXBwLWVkZ2Utc3NyJTJGcGFnZSZwYWdlPSUyRmFwcC1lZGdlLXNzciUyRnBhZ2UmcGFnZVBhdGg9cHJpdmF0ZS1uZXh0LWFwcC1kaXIlMkZhcHAtZWRnZS1zc3IlMkZwYWdlLmpzJmFwcERpcj0lMkZ0bXAlMkZuZXh0LXN0YXRzOU9JV0xQJTJGc3RhdHMtYXBwJTJGYXBwJmFwcFBhdGhzPSUyRmFwcC1lZGdlLXNzciUyRnBhZ2UmcGFnZUV4dGVuc2lvbnM9dHN4JnBhZ2VFeHRlbnNpb25zPXRzJnBhZ2VFeHRlbnNpb25zPWpzeCZwYWdlRXh0ZW5zaW9ucz1qcyZiYXNlUGF0aD0mYXNzZXRQcmVmaXg9Jm5leHRDb25maWdPdXRwdXQ9JnByZWZlcnJlZFJlZ2lvbj0mbWlkZGxld2FyZUNvbmZpZz1lMzAlM0Qh","sriEnabled":false,"middlewareConfig":"e30="}!
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/esm/lib/page-types.js
+      var page_types = __webpack_require__(3431);
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/esm/server/app-render/encryption-utils.js
+      var encryption_utils = __webpack_require__(5289);
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/esm/server/app-render/action-utils.js
+      var action_utils = __webpack_require__(1863); // CONCATENATED MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/dist/build/webpack/loaders/next-edge-ssr-loader/index.js?{"absolute500Path":"","absoluteAppPath":"next/dist/pages/_app","absoluteDocumentPath":"next/dist/pages/_document","absoluteErrorPath":"next/dist/pages/_error","absolutePagePath":"private-next-app-dir/app-edge-ssr/page.js","dev":false,"isServerComponent":true,"page":"/app-edge-ssr/page","stringifiedConfig":"eyJlbnYiOnt9LCJlc2xpbnQiOnsiaWdub3JlRHVyaW5nQnVpbGRzIjpmYWxzZX0sInR5cGVzY3JpcHQiOnsiaWdub3JlQnVpbGRFcnJvcnMiOmZhbHNlLCJ0c2NvbmZpZ1BhdGgiOiJ0c2NvbmZpZy5qc29uIn0sImRpc3REaXIiOiIubmV4dCIsImNsZWFuRGlzdERpciI6dHJ1ZSwiYXNzZXRQcmVmaXgiOiIiLCJjYWNoZU1heE1lbW9yeVNpemUiOjUyNDI4ODAwLCJjb25maWdPcmlnaW4iOiJuZXh0LmNvbmZpZy5qcyIsInVzZUZpbGVTeXN0ZW1QdWJsaWNSb3V0ZXMiOnRydWUsImdlbmVyYXRlRXRhZ3MiOnRydWUsInBhZ2VFeHRlbnNpb25zIjpbInRzeCIsInRzIiwianN4IiwianMiXSwicG93ZXJlZEJ5SGVhZGVyIjp0cnVlLCJjb21wcmVzcyI6dHJ1ZSwiaW1hZ2VzIjp7ImRldmljZVNpemVzIjpbNjQwLDc1MCw4MjgsMTA4MCwxMjAwLDE5MjAsMjA0OCwzODQwXSwiaW1hZ2VTaXplcyI6WzE2LDMyLDQ4LDY0LDk2LDEyOCwyNTYsMzg0XSwicGF0aCI6Ii9fbmV4dC9pbWFnZSIsImxvYWRlciI6ImRlZmF1bHQiLCJsb2FkZXJGaWxlIjoiIiwiZG9tYWlucyI6W10sImRpc2FibGVTdGF0aWNJbWFnZXMiOmZhbHNlLCJtaW5pbXVtQ2FjaGVUVEwiOjYwLCJmb3JtYXRzIjpbImltYWdlL3dlYnAiXSwiZGFuZ2Vyb3VzbHlBbGxvd1NWRyI6ZmFsc2UsImNvbnRlbnRTZWN1cml0eVBvbGljeSI6InNjcmlwdC1zcmMgJ25vbmUnOyBmcmFtZS1zcmMgJ25vbmUnOyBzYW5kYm94OyIsImNvbnRlbnREaXNwb3NpdGlvblR5cGUiOiJhdHRhY2htZW50IiwicmVtb3RlUGF0dGVybnMiOltdLCJ1bm9wdGltaXplZCI6ZmFsc2V9LCJkZXZJbmRpY2F0b3JzIjp7ImJ1aWxkQWN0aXZpdHkiOnRydWUsImJ1aWxkQWN0aXZpdHlQb3NpdGlvbiI6ImJvdHRvbS1yaWdodCJ9LCJvbkRlbWFuZEVudHJpZXMiOnsibWF4SW5hY3RpdmVBZ2UiOjYwMDAwLCJwYWdlc0J1ZmZlckxlbmd0aCI6NX0sImFtcCI6eyJjYW5vbmljYWxCYXNlIjoiIn0sImJhc2VQYXRoIjoiIiwic2Fzc09wdGlvbnMiOnt9LCJ0cmFpbGluZ1NsYXNoIjpmYWxzZSwiaTE4biI6bnVsbCwicHJvZHVjdGlvbkJyb3dzZXJTb3VyY2VNYXBzIjpmYWxzZSwib3B0aW1pemVGb250cyI6dHJ1ZSwiZXhjbHVkZURlZmF1bHRNb21lbnRMb2NhbGVzIjp0cnVlLCJzZXJ2ZXJSdW50aW1lQ29uZmlnIjp7fSwicHVibGljUnVudGltZUNvbmZpZyI6e30sInJlYWN0UHJvZHVjdGlvblByb2ZpbGluZyI6ZmFsc2UsInJlYWN0U3RyaWN0TW9kZSI6bnVsbCwiaHR0cEFnZW50T3B0aW9ucyI6eyJrZWVwQWxpdmUiOnRydWV9LCJzdGF0aWNQYWdlR2VuZXJhdGlvblRpbWVvdXQiOjYwLCJtb2R1bGFyaXplSW1wb3J0cyI6eyJAbXVpL2ljb25zLW1hdGVyaWFsIjp7InRyYW5zZm9ybSI6IkBtdWkvaWNvbnMtbWF0ZXJpYWwve3ttZW1iZXJ9fSJ9LCJsb2Rhc2giOnsidHJhbnNmb3JtIjoibG9kYXNoL3t7bWVtYmVyfX0ifX0sImV4cGVyaW1lbnRhbCI6eyJmbHlpbmdTaHV0dGxlIjpmYWxzZSwicHJlcmVuZGVyRWFybHlFeGl0Ijp0cnVlLCJzZXJ2ZXJNaW5pZmljYXRpb24iOnRydWUsInNlcnZlclNvdXJjZU1hcHMiOmZhbHNlLCJsaW5rTm9Ub3VjaFN0YXJ0IjpmYWxzZSwiY2FzZVNlbnNpdGl2ZVJvdXRlcyI6ZmFsc2UsInByZWxvYWRFbnRyaWVzT25TdGFydCI6dHJ1ZSwiY2xpZW50Um91dGVyRmlsdGVyIjp0cnVlLCJjbGllbnRSb3V0ZXJGaWx0ZXJSZWRpcmVjdHMiOmZhbHNlLCJmZXRjaENhY2hlS2V5UHJlZml4IjoiIiwibWlkZGxld2FyZVByZWZldGNoIjoiZmxleGlibGUiLCJvcHRpbWlzdGljQ2xpZW50Q2FjaGUiOnRydWUsIm1hbnVhbENsaWVudEJhc2VQYXRoIjpmYWxzZSwiY3B1cyI6MTksIm1lbW9yeUJhc2VkV29ya2Vyc0NvdW50IjpmYWxzZSwiaXNyRmx1c2hUb0Rpc2siOnRydWUsIndvcmtlclRocmVhZHMiOmZhbHNlLCJvcHRpbWl6ZUNzcyI6ZmFsc2UsIm5leHRTY3JpcHRXb3JrZXJzIjpmYWxzZSwic2Nyb2xsUmVzdG9yYXRpb24iOmZhbHNlLCJleHRlcm5hbERpciI6ZmFsc2UsImRpc2FibGVPcHRpbWl6ZWRMb2FkaW5nIjpmYWxzZSwiZ3ppcFNpemUiOnRydWUsImNyYUNvbXBhdCI6ZmFsc2UsImVzbUV4dGVybmFscyI6dHJ1ZSwiZnVsbHlTcGVjaWZpZWQiOmZhbHNlLCJvdXRwdXRGaWxlVHJhY2luZ1Jvb3QiOiIvdG1wL25leHQtc3RhdHM5T0lXTFAvc3RhdHMtYXBwIiwic3djVHJhY2VQcm9maWxpbmciOmZhbHNlLCJmb3JjZVN3Y1RyYW5zZm9ybXMiOmZhbHNlLCJsYXJnZVBhZ2VEYXRhQnl0ZXMiOjEyODAwMCwiYWRqdXN0Rm9udEZhbGxiYWNrcyI6ZmFsc2UsImFkanVzdEZvbnRGYWxsYmFja3NXaXRoU2l6ZUFkanVzdCI6ZmFsc2UsInR5cGVkUm91dGVzIjpmYWxzZSwiaW5zdHJ1bWVudGF0aW9uSG9vayI6ZmFsc2UsInBhcmFsbGVsU2VydmVyQ29tcGlsZXMiOmZhbHNlLCJwYXJhbGxlbFNlcnZlckJ1aWxkVHJhY2VzIjpmYWxzZSwicHByIjpmYWxzZSwid2VicGFja01lbW9yeU9wdGltaXphdGlvbnMiOmZhbHNlLCJvcHRpbWl6ZVNlcnZlclJlYWN0Ijp0cnVlLCJ1c2VFYXJseUltcG9ydCI6ZmFsc2UsInN0YWxlVGltZXMiOnsiZHluYW1pYyI6MCwic3RhdGljIjozMDB9LCJhZnRlciI6ZmFsc2UsIm9wdGltaXplUGFja2FnZUltcG9ydHMiOlsibHVjaWRlLXJlYWN0IiwiZGF0ZS1mbnMiLCJsb2Rhc2gtZXMiLCJyYW1kYSIsImFudGQiLCJyZWFjdC1ib290c3RyYXAiLCJhaG9va3MiLCJAYW50LWRlc2lnbi9pY29ucyIsIkBoZWFkbGVzc3VpL3JlYWN0IiwiQGhlYWRsZXNzdWktZmxvYXQvcmVhY3QiLCJAaGVyb2ljb25zL3JlYWN0LzIwL3NvbGlkIiwiQGhlcm9pY29ucy9yZWFjdC8yNC9zb2xpZCIsIkBoZXJvaWNvbnMvcmVhY3QvMjQvb3V0bGluZSIsIkB2aXN4L3Zpc3giLCJAdHJlbW9yL3JlYWN0IiwicnhqcyIsIkBtdWkvbWF0ZXJpYWwiLCJAbXVpL2ljb25zLW1hdGVyaWFsIiwicmVjaGFydHMiLCJyZWFjdC11c2UiLCJlZmZlY3QiLCJAZWZmZWN0L3NjaGVtYSIsIkBlZmZlY3QvcGxhdGZvcm0iLCJAZWZmZWN0L3BsYXRmb3JtLW5vZGUiLCJAZWZmZWN0L3BsYXRmb3JtLWJyb3dzZXIiLCJAZWZmZWN0L3BsYXRmb3JtLWJ1biIsIkBlZmZlY3Qvc3FsIiwiQGVmZmVjdC9zcWwtbXNzcWwiLCJAZWZmZWN0L3NxbC1teXNxbDIiLCJAZWZmZWN0L3NxbC1wZyIsIkBlZmZlY3Qvc3FsLXNxdWxpdGUtbm9kZSIsIkBlZmZlY3Qvc3FsLXNxdWxpdGUtYnVuIiwiQGVmZmVjdC9zcWwtc3F1bGl0ZS13YXNtIiwiQGVmZmVjdC9zcWwtc3F1bGl0ZS1yZWFjdC1uYXRpdmUiLCJAZWZmZWN0L3JwYyIsIkBlZmZlY3QvcnBjLWh0dHAiLCJAZWZmZWN0L3R5cGVjbGFzcyIsIkBlZmZlY3QvZXhwZXJpbWVudGFsIiwiQGVmZmVjdC9vcGVudGVsZW1ldHJ5IiwiQG1hdGVyaWFsLXVpL2NvcmUiLCJAbWF0ZXJpYWwtdWkvaWNvbnMiLCJAdGFibGVyL2ljb25zLXJlYWN0IiwibXVpLWNvcmUiLCJyZWFjdC1pY29ucy9haSIsInJlYWN0LWljb25zL2JpIiwicmVhY3QtaWNvbnMvYnMiLCJyZWFjdC1pY29ucy9jZyIsInJlYWN0LWljb25zL2NpIiwicmVhY3QtaWNvbnMvZGkiLCJyZWFjdC1pY29ucy9mYSIsInJlYWN0LWljb25zL2ZhNiIsInJlYWN0LWljb25zL2ZjIiwicmVhY3QtaWNvbnMvZmkiLCJyZWFjdC1pY29ucy9naSIsInJlYWN0LWljb25zL2dvIiwicmVhY3QtaWNvbnMvZ3IiLCJyZWFjdC1pY29ucy9oaSIsInJlYWN0LWljb25zL2hpMiIsInJlYWN0LWljb25zL2ltIiwicmVhY3QtaWNvbnMvaW8iLCJyZWFjdC1pY29ucy9pbzUiLCJyZWFjdC1pY29ucy9saWEiLCJyZWFjdC1pY29ucy9saWIiLCJyZWFjdC1pY29ucy9sdSIsInJlYWN0LWljb25zL21kIiwicmVhY3QtaWNvbnMvcGkiLCJyZWFjdC1pY29ucy9yaSIsInJlYWN0LWljb25zL3J4IiwicmVhY3QtaWNvbnMvc2kiLCJyZWFjdC1pY29ucy9zbCIsInJlYWN0LWljb25zL3RiIiwicmVhY3QtaWNvbnMvdGZpIiwicmVhY3QtaWNvbnMvdGkiLCJyZWFjdC1pY29ucy92c2MiLCJyZWFjdC1pY29ucy93aSJdfSwiYnVuZGxlUGFnZXNSb3V0ZXJEZXBlbmRlbmNpZXMiOmZhbHNlLCJjb25maWdGaWxlIjoiL3RtcC9uZXh0LXN0YXRzOU9JV0xQL3N0YXRzLWFwcC9uZXh0LmNvbmZpZy5qcyIsImNvbmZpZ0ZpbGVOYW1lIjoibmV4dC5jb25maWcuanMifQ==","pagesType":"app","appDirLoader":"bmV4dC1hcHAtbG9hZGVyP25hbWU9YXBwJTJGYXBwLWVkZ2Utc3NyJTJGcGFnZSZwYWdlPSUyRmFwcC1lZGdlLXNzciUyRnBhZ2UmcGFnZVBhdGg9cHJpdmF0ZS1uZXh0LWFwcC1kaXIlMkZhcHAtZWRnZS1zc3IlMkZwYWdlLmpzJmFwcERpcj0lMkZ0bXAlMkZuZXh0LXN0YXRzOU9JV0xQJTJGc3RhdHMtYXBwJTJGYXBwJmFwcFBhdGhzPSUyRmFwcC1lZGdlLXNzciUyRnBhZ2UmcGFnZUV4dGVuc2lvbnM9dHN4JnBhZ2VFeHRlbnNpb25zPXRzJnBhZ2VFeHRlbnNpb25zPWpzeCZwYWdlRXh0ZW5zaW9ucz1qcyZiYXNlUGF0aD0mYXNzZXRQcmVmaXg9Jm5leHRDb25maWdPdXRwdXQ9JnByZWZlcnJlZFJlZ2lvbj0mbWlkZGxld2FyZUNvbmZpZz1lMzAlM0Qh","sriEnabled":false,"middlewareConfig":"e30="}!
       var _self___RSC_MANIFEST;
 
       const incrementalCacheHandler = null;
@@ -426,47 +426,47 @@
       /***/
     },
 
-    /***/ 5575: /***/ (
+    /***/ 6975: /***/ (
       __unused_webpack_module,
       __unused_webpack_exports,
       __webpack_require__
     ) => {
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.bind(__webpack_require__, 2300)
+        __webpack_require__.bind(__webpack_require__, 7156)
       );
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.bind(__webpack_require__, 3447)
+        __webpack_require__.bind(__webpack_require__, 209)
       );
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.bind(__webpack_require__, 4596)
+        __webpack_require__.bind(__webpack_require__, 3612)
       );
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.bind(__webpack_require__, 5542)
+        __webpack_require__.bind(__webpack_require__, 6231)
       );
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.bind(__webpack_require__, 3314)
+        __webpack_require__.bind(__webpack_require__, 8106)
       );
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.bind(__webpack_require__, 9654)
+        __webpack_require__.bind(__webpack_require__, 6002)
       );
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.bind(__webpack_require__, 9870)
+        __webpack_require__.bind(__webpack_require__, 5585)
       );
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.bind(__webpack_require__, 1713)
+        __webpack_require__.bind(__webpack_require__, 7135)
       );
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.bind(__webpack_require__, 959)
+        __webpack_require__.bind(__webpack_require__, 579)
       );
 
       /***/
     },
 
-    /***/ 3158: /***/ () => {
+    /***/ 3906: /***/ () => {
       /***/
     },
 
-    /***/ 2287: /***/ (
+    /***/ 8236: /***/ (
       __unused_webpack_module,
       __webpack_exports__,
       __webpack_require__
@@ -486,7 +486,7 @@
       /***/
     },
 
-    /***/ 8369: /***/ (
+    /***/ 2716: /***/ (
       __unused_webpack_module,
       __webpack_exports__,
       __webpack_require__
@@ -498,7 +498,7 @@
         /* harmony export */
       });
       /* harmony import */ var react_jsx_runtime__WEBPACK_IMPORTED_MODULE_0__ =
-        __webpack_require__(3046);
+        __webpack_require__(5868);
 
       function RootLayout({ children }) {
         return /*#__PURE__*/ (0,
@@ -517,7 +517,7 @@
     // webpackRuntimeModules
     /******/ var __webpack_exec__ = (moduleId) =>
       __webpack_require__((__webpack_require__.s = moduleId));
-    /******/ __webpack_require__.O(0, [374, 641], () => __webpack_exec__(3080));
+    /******/ __webpack_require__.O(0, [19, 784], () => __webpack_exec__(2444));
     /******/ var __webpack_exports__ = __webpack_require__.O();
     /******/ (_ENTRIES = typeof _ENTRIES === "undefined" ? {} : _ENTRIES)[
       "middleware_app/app-edge-ssr/page"
```
</details>
<details>
<summary>Diff for <strong>middleware.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>edge-ssr.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>image-HASH.js</strong></summary>

```diff
@@ -1,7 +1,7 @@
 (self["webpackChunk_N_E"] = self["webpackChunk_N_E"] || []).push([
   [8358],
   {
-    /***/ 4176: /***/ (
+    /***/ 5930: /***/ (
       __unused_webpack_module,
       __unused_webpack_exports,
       __webpack_require__
@@ -9,7 +9,7 @@
       (window.__NEXT_P = window.__NEXT_P || []).push([
         "/image",
         function () {
-          return __webpack_require__(2010);
+          return __webpack_require__(3052);
         },
       ]);
       if (false) {
@@ -18,7 +18,7 @@
       /***/
     },
 
-    /***/ 9626: /***/ (module, exports, __webpack_require__) => {
+    /***/ 1892: /***/ (module, exports, __webpack_require__) => {
       "use strict";
       /* __next_internal_client_entry_do_not_use__  cjs */
       Object.defineProperty(exports, "__esModule", {
@@ -40,15 +40,15 @@
         __webpack_require__(7141)
       );
       const _head = /*#__PURE__*/ _interop_require_default._(
-        __webpack_require__(8940)
+        __webpack_require__(1951)
       );
-      const _getimgprops = __webpack_require__(1395);
-      const _imageconfig = __webpack_require__(2568);
-      const _imageconfigcontextsharedruntime = __webpack_require__(9067);
-      const _warnonce = __webpack_require__(3082);
-      const _routercontextsharedruntime = __webpack_require__(9818);
+      const _getimgprops = __webpack_require__(6033);
+      const _imageconfig = __webpack_require__(9782);
+      const _imageconfigcontextsharedruntime = __webpack_require__(3392);
+      const _warnonce = __webpack_require__(6433);
+      const _routercontextsharedruntime = __webpack_require__(3918);
       const _imageloader = /*#__PURE__*/ _interop_require_default._(
-        __webpack_require__(3978)
+        __webpack_require__(646)
       );
       // This is replaced by webpack define plugin
       const configEnv = {
@@ -376,7 +376,7 @@
       /***/
     },
 
-    /***/ 1395: /***/ (
+    /***/ 6033: /***/ (
       __unused_webpack_module,
       exports,
       __webpack_require__
@@ -392,9 +392,9 @@
           return getImgProps;
         },
       });
-      const _warnonce = __webpack_require__(3082);
-      const _imageblursvg = __webpack_require__(8846);
-      const _imageconfig = __webpack_require__(2568);
+      const _warnonce = __webpack_require__(6433);
+      const _imageblursvg = __webpack_require__(6547);
+      const _imageconfig = __webpack_require__(9782);
       const VALID_LOADING_VALUES =
         /* unused pure expression or super */ null && [
           "lazy",
@@ -769,7 +769,7 @@
       /***/
     },
 
-    /***/ 8846: /***/ (__unused_webpack_module, exports) => {
+    /***/ 6547: /***/ (__unused_webpack_module, exports) => {
       "use strict";
       /**
        * A shared function, used on both client and server, to generate a SVG blur placeholder.
@@ -824,7 +824,7 @@
       /***/
     },
 
-    /***/ 5626: /***/ (
+    /***/ 9915: /***/ (
       __unused_webpack_module,
       exports,
       __webpack_require__
@@ -851,10 +851,10 @@
         },
       });
       const _interop_require_default = __webpack_require__(1478);
-      const _getimgprops = __webpack_require__(1395);
-      const _imagecomponent = __webpack_require__(9626);
+      const _getimgprops = __webpack_require__(6033);
+      const _imagecomponent = __webpack_require__(1892);
       const _imageloader = /*#__PURE__*/ _interop_require_default._(
-        __webpack_require__(3978)
+        __webpack_require__(646)
       );
       function getImageProps(imgProps) {
         const { props } = (0, _getimgprops.getImgProps)(imgProps, {
@@ -886,7 +886,7 @@
       /***/
     },
 
-    /***/ 3978: /***/ (__unused_webpack_module, exports) => {
+    /***/ 646: /***/ (__unused_webpack_module, exports) => {
       "use strict";
 
       Object.defineProperty(exports, "__esModule", {
@@ -921,7 +921,7 @@
       /***/
     },
 
-    /***/ 2010: /***/ (
+    /***/ 3052: /***/ (
       __unused_webpack_module,
       __webpack_exports__,
       __webpack_require__
@@ -938,8 +938,8 @@
 
       // EXTERNAL MODULE: ./node_modules/.pnpm/react@19.0.0-rc-f994737d14-20240522/node_modules/react/jsx-runtime.js
       var jsx_runtime = __webpack_require__(5134);
-      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+main-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_fxmik4ojtenl5wvndfh5uokf6a/node_modules/next/image.js
-      var next_image = __webpack_require__(7365);
+      // EXTERNAL MODULE: ./node_modules/.pnpm/file+..+diff-repo+packages+next+next-packed.tgz_react-dom@19.0.0-rc-f994737d14-20240522_react_3yjmd2pnnuuncmihyqifk2hdsq/node_modules/next/image.js
+      var next_image = __webpack_require__(7027);
       var image_default = /*#__PURE__*/ __webpack_require__.n(next_image); // CONCATENATED MODULE: ./pages/nextjs.png
       /* harmony default export */ const nextjs = {
         src: "/_next/static/media/nextjs.cae0b805.png",
@@ -969,12 +969,12 @@
       /***/
     },
 
-    /***/ 7365: /***/ (
+    /***/ 7027: /***/ (
       module,
       __unused_webpack_exports,
       __webpack_require__
     ) => {
-      module.exports = __webpack_require__(5626);
+      module.exports = __webpack_require__(9915);
 
       /***/
     },
@@ -984,7 +984,7 @@
     /******/ var __webpack_exec__ = (moduleId) =>
       __webpack_require__((__webpack_require__.s = moduleId));
     /******/ __webpack_require__.O(0, [2888, 9774, 179], () =>
-      __webpack_exec__(4176)
+      __webpack_exec__(5930)
     );
     /******/ var __webpack_exports__ = __webpack_require__.O();
     /******/ _N_E = __webpack_exports__;
```
</details>
<details>
<summary>Diff for <strong>2358-HASH.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>webpack-HASH.js</strong></summary>

```diff
@@ -205,7 +205,7 @@
     /******/ __webpack_require__.u = (chunkId) => {
       /******/ // return url for filenames based on template
       /******/ return (
-        "static/chunks/" + chunkId + "." + "f31fb2ff6a6c8e70" + ".js"
+        "static/chunks/" + chunkId + "." + "0844cc9f070434a0" + ".js"
       );
       /******/
     };
@@ -223,22 +223,6 @@
     /******/
   })();
   /******/
-  /******/ /* webpack/runtime/global */
-  /******/ (() => {
-    /******/ __webpack_require__.g = (function () {
-      /******/ if (typeof globalThis === "object") return globalThis;
-      /******/ try {
-        /******/ return this || new Function("return this")();
-        /******/
-      } catch (e) {
-        /******/ if (typeof window === "object") return window;
-        /******/
-      }
-      /******/
-    })();
-    /******/
-  })();
-  /******/
   /******/ /* webpack/runtime/hasOwnProperty shorthand */
   /******/ (() => {
     /******/ __webpack_require__.o = (obj, prop) =>
```
</details>
<details>
<summary>Diff for <strong>app-page-exp..ntime.dev.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>app-page-exp..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.dev.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>app-page.runtime.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..ntime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.ru..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-api-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-api.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-api.ru..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-turbo...time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages.runtime.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>server.runtime.prod.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 66360784626d0f66caa208a5048b9e8583450a4f

gnoff commented:
@lfades this is ambitious! I previously failed to do something similar but ended up making a fixture to assert various vendoring strategies would work across all different kinds of package managers.

One of the problems with our tests is they produce tarballs and package managers don't actually install from tarballs with the exact same semantics as they do with package names/urls. This caused my tests to all pass while failing in the real world after publishing.

I made this little test bench and came up with a strategy that works for all major package managers and tested using veredaccio to simulate a real package install. I haven't documented it well but LMK if you want to discuss it and see if it is relevant to this work

https://github.com/gnoff/vendor-strategy

lfades commented:
@gnoff After finishing most of the work of implementing `@next/vendored` I'm actually not that happy with it. There's a considerable amount of breaking changes which make this PR very hard to review. My first iteration was to use a nested `node_modules` instead of `@next/vendored` as that would make the transition easier for all the internal stuff that depends on those packages being inside the next package while still allowing the `next` package to be compatible with package.json `exports`, which was my initial motivation in making all of this.

But honestly this looks like a nightmare to review so I'm more inclined into closing the PR and maybe try again with incremental changes. I'm also considering that instead of having a `@next/vendored` package we could instead have one package for every vendored dependency. That should make it easier to handle the types and exports of every package, by more closely matching how they work in npm (and automate all of this).

lfades commented:
Closing the PR for now, this is too big of a change to test properly in a single PR. It could be nice to have this changes but with a better plan where this can be properly reviewed.
